posts/2024-04-29-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,418 @@
+
+Eclipse JNoSQL covers four NoSQL database types: *key-value*, *column*, *document* and *graph* databases.
+
+[NOTE]
+Currently, the Jakarta NoSQL doesn't define an API for Graph database types but Eclipse JNoSQL provides a Graph template to explore the specific behavior of this database type by using https://tinkerpop.apache.org/[Apache TinkerPop] as a communication layer.
+
+As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+==== Why Eclipse JNoSQL?
+
+In the code samples below, note the APIs similarities and differences accross different vendors. All samples demonstrate a commonly used functionality for Document databases: a document creation and the addition of a property to that document:
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/mongodb.png[MongoDB ,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new Document();
+document.append(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/ArangoDB.png[Arango DB,align="left" width=25%,height=25%]
+[source, java]
+----
+var baseDocument = new BaseDocument();
+baseDocument.addAttribute(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/couchbase.svg[Couchbase,align="left" width=25%,height=25%]
+[source, java]
+----
+var jsonObject = JsonObject.create();
+jsonObject.put(name, value);
+----
+****
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/img/logos/orientdb.png[Orient Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var document = new ODocument("collection");
+document.field(name, value);
+----
+****
+
+With Eclipse JNoSQL, developers can use a common API to integrate with different database types, free of vendor lock-in, and with a low cognitive load during learning phases. For example, using the Document API, it's possible to switch between MongoDB and ArangoDB as needed, based on Convention Over Configuration (CoC).
+
+[.img_border_light]
+****
+image::https://jnosql.github.io/images/home_logo.png[Eclipse JNoSQL Project,align="left" width=25%,height=25%]
+[source, java]
+----
+var entity = CommunicationEntity.of("collection");
+entity.add(name, value);
+----
+****
+
+Check out some of the Jakarta NoSQL annotations:
+
+[source, java]
+----
+import jakarta.nosql.Entity;
+import jakarta.nosql.Id;
+import jakarta.nosql.Column;
+
+@Entity
+public class Book {
+
+    @Id
+    private String isbn;
+
+    @Column
+    private String title;
+
+    @Column
+    private String author;
+
+    @Convert(YearConverter.class)
+    @Column
+    private Year year;
+
+}
+----
+
+When using Java 17+, Eclipse JNoSQL allows the usage of Java Records as entities:
```

Comment:
Similar comment for this one too - would be nice to maybe have more linkage to the section before